apps/desktop/src/autofill/services/desktop-fido2-user-interface.service.ts

```diff
@@ -71,36 +88,90 @@ export class DesktopFido2UserInterfaceSession implements Fido2UserInterfaceSessi
     private cipherService: CipherService,
     private accountService: AccountService,
     private logService: LogService,
-    private messagingService: MessagingService,
     private router: Router,
     private desktopSettingsService: DesktopSettingsService,
+    private windowObject: NativeWindowObject,
   ) {}
 
   private confirmCredentialSubject = new Subject<boolean>();
   private createdCipher: Cipher;
 
+  private availableCipherIdsSubject = new BehaviorSubject<string[]>(null);
+  /**
+   * Observable that emits available cipher IDs once they're confirmed by the UI
+   */
+  get availableCipherIds$(): Observable<string[]> {
+    return this.availableCipherIdsSubject.pipe(
+      filter((ids) => ids != null),
+      take(1),
+    );
+  }
```

Comment:
issue: don't use getters which will create new observables for every single change-detection call causing massive performance penalties
```suggestion
  /**
   * Observable that emits available cipher IDs once they're confirmed by the UI
   */
  availableCipherIds$ = this.availableCipherIdsSubject.pipe(
    filter((ids) => ids != null),
    take(1),
  );
```